src/env-inl.h

```diff
@@ -538,16 +550,27 @@ inline void Environment::set_fs_stats_field_array(double* fields) {
 
 void Environment::SetImmediate(native_immediate_callback cb,
                                void* data,
-                               v8::Local<v8::Object> obj) {
+                               v8::Local<v8::Object> obj,
+                               bool ref) {
```

Comment:
Rather than a `bool`, I'd prefer an enum type argument here.

Reply:
I didn't want that argument to be used, to be honest (it was solely there to not have to replicate the code within `SetUnrefImmediate`). Would something like this address the concern (with `CreateImmediate` being private)? Or would you still prefer enum in this situation?

```cpp
void Environment::CreateImmediate(native_immediate_callback cb,
                               void* data,
                               v8::Local<v8::Object> obj,
                               bool ref) {
  native_immediate_callbacks_.push_back({
    cb,
    data,
    std::unique_ptr<v8::Persistent<v8::Object>>(obj.IsEmpty() ?
        nullptr : new v8::Persistent<v8::Object>(isolate_, obj)),
    ref
  });
  immediate_info()->count_inc(1);
  if (ref) {
    if (immediate_info()->ref_count() == 0)
      ToggleImmediateRef(true);
    immediate_info()->ref_count_inc(1);
  }
}

void Environment::SetImmediate(native_immediate_callback cb,
                               void* data,
                               v8::Local<v8::Object> obj) {
  CreateImmediate(cb, data, obj, true);
}

void Environment::SetUnrefImmediate(native_immediate_callback cb,
                                    void* data,
                                    v8::Local<v8::Object> obj) {
  CreateImmediate(cb, data, obj, false);
}
```

Reply:
Since `CreateImmediate()` is private, it's not a big deal. Just a preference.